thrust/cmake/ThrustHeaderTesting.cmake

```diff
@@ -124,6 +124,9 @@ function(thrust_add_header_test thrust_target label definitions)
     target_compile_definitions(${headertest_target} PRIVATE CCCL_IGNORE_HEADER_MACRO_CHECKS)
   endif()
 
+  # We are deprecating some thrust features, so disable deprecation warnings
+  target_compile_definitions(${headertest_target} PRIVATE CCCL_IGNORE_DEPRECATED_API)
```

Comment:
Globally disabling deprecation warnings is not a good idea. We've never needed to do this before.

Deprecation warnings are usually only emitted when we're using a deprecated utility in our own implementations. We should narrowly scope any necessary warning suppressions.

Reply:
Those are only for the header tests, which verify that we can build our own headers.

I believe we should suppress any warning that comes from CCCL here.

Note that we **do not** suppress any warnings that are coming from outside of CCCL

Reply:
The header tests are to ensure that our headers build cleanly for our users. If we need to suppress deprecation warnings just from including them, so will our users.

We've never had to blanket exclude deprecation warnings for any project's tests, examples, or headers before and we've been through many rounds of deprecations. If a suppression is truly needed here it can be done at a much more narrow scope.

Reply:
Will limit it to the affected header only